content/tutorials/react-native-presence-publish-subscribe.textile

```diff
@@ -0,0 +1,539 @@
+
+
+h2. Setting Up React Native
+
+To get started first, you need to install the @React Native CLI@ if you don’t already have it installed. To install the @CLI@, run:
+
+```[javascript]
+ npm install -g react-native-cli
+```
+
+After installing the @CLI@, it’s time to create your project. Open a terminal, create a new project called @ably-presence-publish-subscribe-react-native@:
+
+```[javascript]
+react-native init ably-presence-publish-subscribe-react-native
+```
+
+Before I continue, let me explain why I named the project that way.
+
+# ably: I will use the @ably@ real-time library for real-time interactions
+# presence: I will cover the concept of ably presence channels
+# publish: I will cover the concept of publishing messages in ably
+# subscribe: I will cover the concept of subscribing to channels and events in ably
+# react-native: I will build the chat application using @React-Native@ 
+
+Note: You can give your project any other name you wish to.
+
+Wait until React Native does all its installations, then you can change directory into the new project and run the application:
+
+```[sh]
+//change directory to ably-presence-publish-subscribe-react-native
+cd ably-presence-publish-subscribe-react-native
+//run the application for android
+react-native run-android
+//run the application for ios
+react-native run-ios
+```
+
+If all goes well, you should see a screen that says “Welcome to React Native” come up on your emulator or device.
+
+
+h2. Setting Up Ably
+
+You need to sign up on "Ably":https://www.ably.io/, create a new app, click on the @manage app@ button at the left-hand side of the app, click on the @Api keys@  tab and copy our Api Key.
+
+<a href="/images/tutorials/copy-api-key.png" target="_blank">
+  <img src="/images/tutorials/copy-api-key.png" style="width: 100%" alt="Copy API Key screenshot">
+</a>
+
+Next, you will install the required libraries for this app. 
+
+```[sh]
+npm install ably ably-react-native express body-parser 
+```
+In the above bash command, you installed 4 packages. I will explain what the four packages are used for:
+
+# express: This is a Node.js web framework which we’ll use to create our API.
+# body-parser: This library is used by Express to parse body requests.
+# ably: This library is the official @Ably@ library for Node js
+# ably-react-native: This library is the official @Ably@ library for React Native
+
+
+h2. Creating Our API
+
+First, Create a new file called @server.js@, which serves as your API in our root folder and paste the following:
+
+```[javascript]
+//require express
+var express = require('express')
+//define app as in instance of express
+var app = express()
+//require bosy-parser
+var bodyParser = require('body-parser')
+
+//use bodyparser as a middle ware
+app.use(bodyParser.json())
+//ably realtime
+const ably = require('ably').Realtime;
+// instantiate an instance of ably passing in our API key
+const myably = new ably('XXX_API_KEY')
```

Comment:
`myably` is a pretty weird name.  Why not something like:

```
const Ably = require('ably');
const ablyRest = new Ably.Rest('[INSERT_YOUR_API_KEY_HERE]');
```

Also, why are you using semicolons sometimes and not others?  Can you please be consistent

Reply:
I have updated my code to follow through with semicolons